tests/pfcwd/files/pfcwd_helper.py

```diff
@@ -364,3 +366,46 @@ def fetch_vendor_specific_diagnosis_re(duthost):
         return ""
 
     return VENDOR_SPEC_ADDITIONAL_INFO_RE.get(duthost.facts["asic_type"], "")
+
+def is_pfcwd_port_stormed( dut, storm_port ):
+   ''' storm_queue: Ethernet<num>:<queue_num>'''
+   stats = pfcwd_stats( dut )
+   for queue, status in stats.items():
+      if queue.startswith( storm_port ) and status == 'stormed':
+         return True
+   return False
+
+def pfcwd_stats( dut ):
+   result = dut.shell( "show pfcwd stats", module_ignore_errors=True, verbose=False )
+   if result['rc'] != 0:
+      return {}
+   return parse_pfcwd_stats( result[ 'stdout_lines' ] )
+
+def parse_pfcwd_stats( lines ):
+   stats = {}
+   for line in lines:
+      tokens = line.split()
+      queue = tokens[ 0 ]
+      if queue.startswith( 'Ether' ):
+         status = tokens[ 1 ]
+         stats[ queue ] = status
+   return stats
+
+def wait_until_pfcwd_stormed( dut, storm_port ):
+   is_stormed = wait_until(
+      10*60, 5, 0,
```

Comment:
Maybe we could consider using the original sleep value as the delay parameter to avoid the first try.

Reply:
Sure, although in my testing I have observed instances where even a larger delay will not be enough for storm to be detected, and a retry is needed.